modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1458,6 +1465,38 @@ public IgniteInternalFuture<SnapshotPartitionsVerifyTaskResult> checkSnapshot(
                             return;
                         }
 
+                        if (meta.hasCompressedGroups() && grpIds.keySet().stream().anyMatch(meta::isGroupWithCompresion)) {
+                            try {
+                                File dbPath = kctx0.pdsFolderResolver().resolveFolders().persistentStoreRootPath();
+                                int pageSize = kctx0.config().getDataStorageConfiguration().getPageSize();
+
+                                kctx0.compress().checkPageCompressionSupported(dbPath.toPath(), pageSize);
```

Comment:
This method also checks that FS is hole-punchable, but to check snapshot we can proceed even on FS without this ability, so we can just check compression processor is enabled (method `checkPageCompressionSupported()` without parameters)